sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPlugin.scala

```diff
@@ -172,9 +186,14 @@ object ScalaJSPlugin extends AutoPlugin {
   import autoImport._
 
   /** Logs the current statistics about the global IR cache. */
-  def logIRCacheStats(logger: Logger): Unit = {
-    import ScalaJSPluginInternal.globalIRCache
-    logger.debug("Global IR cache stats: " + globalIRCache.stats.logLine)
+  val irCacheStatsLogger: Def.Initialize[Logger => Unit] = Def.setting {
+    val globalIRCacheBox = scalaJSGlobalIRCacheBox.value
+
+    logger: Logger => {
+      globalIRCacheBox.foreach { globalIRCache =>
+        logger.debug("Global IR cache stats: " + globalIRCache.stats.logLine)
+      }
+    }
```

Comment:
I really don't like this. Can we find a better way to do this? Ideally after every task that users the linker :)